trajopt_ifopt/src/collision_evaluators.cpp

```diff
@@ -0,0 +1,1304 @@
+          }
+        };
+
+        return (!((pair_data[0] + collision_config_.collision_margin_buffer) > r.distance));
+      });
+
+  contact_results.erase(end, contact_results.end());
+}
+
+DiscreteCollisionEvaluator::DiscreteCollisionEvaluator(tesseract_kinematics::ForwardKinematics::ConstPtr manip,
+                                                       tesseract_environment::Environment::ConstPtr env,
+                                                       tesseract_environment::AdjacencyMap::ConstPtr adjacency_map,
+                                                       const Eigen::Isometry3d& world_to_base,
+                                                       const TrajOptCollisionConfig& collision_config,
+                                                       bool dynamic_environment)
+  : CollisionEvaluator(std::move(manip),
+                       std::move(env),
+                       std::move(adjacency_map),
+                       world_to_base,
+                       std::move(collision_config),
+                       dynamic_environment)
+{
+  contact_manager_ = env_->getDiscreteContactManager();
+  contact_manager_->setActiveCollisionObjects(adjacency_map_->getActiveLinkNames());
+  contact_manager_->setCollisionMarginData(collision_config.collision_margin_data);
+}
+
+void DiscreteCollisionEvaluator::CalcCollisions(const std::vector<double>& x,
+                                                tesseract_collision::ContactResultMap& dist_results)
+{
+  const Eigen::Map<const Eigen::VectorXd> joint_vals(x.data(), static_cast<Eigen::Index>(x.size()));
+  CalcCollisions(joint_vals, dist_results);
+}
+
+void DiscreteCollisionEvaluator::CalcCollisions(const Eigen::Ref<const Eigen::VectorXd>& dof_vals,
+                                                tesseract_collision::ContactResultMap& dist_results)
+{
+  tesseract_environment::EnvState::Ptr state = get_state_fn_(manip_->getJointNames(), dof_vals);
+
+  for (const auto& link_name : env_->getActiveLinkNames())
+    contact_manager_->setCollisionObjectsTransform(link_name, state->link_transforms[link_name]);
+
+  contact_manager_->contactTest(dist_results, collision_config_.contact_request);
+
+  for (auto& pair : dist_results)
+  {
+    // Contains the contact distance threshold and coefficient for the given link pair
+    double dist =
+        collision_config_.collision_margin_data.getPairCollisionMarginData(pair.first.first, pair.first.second);
+    double coeff = collision_config_.collision_coeff_data.getPairCollisionCoeff(pair.first.first, pair.first.second);
+    const Eigen::Vector2d data = { dist, coeff };
+    auto end = std::remove_if(
+        pair.second.begin(), pair.second.end(), [&data, this](const tesseract_collision::ContactResult& r) {
+          return (!((data[0] + collision_config_.collision_margin_buffer) > r.distance));
+        });
+    pair.second.erase(end, pair.second.end());
+  }
+}
+
+//////////////////////////////////////////
```

Comment:
Should the commented code below be removed?

Reply:
The commented code is what is left to be implemented and will be kept until then.